neuraxle/base.py

```diff
@@ -1663,6 +1718,33 @@ def teardown(self) -> 'BaseStep':
 
         return self
 
+    def apply(self, method_name: str, *kargs) -> 'BaseStep':
+        """
+        Apply the method name to the pipeline step and all of its children.
+
+        :param method_name: method name that need to be called on all steps
+        :param kargs: any additional arguments to be passed to the method
+        :return:
+        """
+        BaseStep.apply(self, method_name, *kargs)
+        for step in self.values():
+            step.apply(method_name, *kargs)
```

Comment:
step = step.apply() ? 

should we require to save the result as a potentially different object, as we do for fit? I think not, but let's be sure. Please make this clear in the docstring.